src/mqt/predictor/rl/actions.py

```diff
@@ -0,0 +1,445 @@
+        PassType.OPT,
+        [PeepholeOptimise2Q()],
+    )
+)
+
+register_action(
+    DeviceIndependentAction(
+        "CliffordSimp",
+        CompilationOrigin.TKET,
+        PassType.OPT,
+        [CliffordSimp()],
+    )
+)
+
+register_action(
+    DeviceIndependentAction(
+        "FullPeepholeOptimiseCX",
+        CompilationOrigin.TKET,
+        PassType.OPT,
+        [FullPeepholeOptimise()],
+    )
+)
+
+register_action(
+    DeviceIndependentAction(
+        "RemoveRedundancies",
+        CompilationOrigin.TKET,
+        PassType.OPT,
+        [RemoveRedundancies()],
+    )
+)
+
+register_action(
+    DeviceDependentAction(
+        "QiskitO3",
+        CompilationOrigin.QISKIT,
+        PassType.OPT,
+        transpile_pass=lambda native_gate, coupling_map: [
+            Collect2qBlocks(),
+            ConsolidateBlocks(basis_gates=native_gate),
+            UnitarySynthesis(basis_gates=native_gate, coupling_map=coupling_map),
+            Optimize1qGatesDecomposition(basis=native_gate),
+            CommutativeCancellation(basis_gates=native_gate),
+            GatesInBasis(native_gate),
+            ConditionalController(
+                common.generate_translation_passmanager(
+                    target=None, basis_gates=native_gate, coupling_map=coupling_map
+                ).to_flow_controller(),
+                condition=lambda property_set: not property_set["all_gates_in_basis"],
+            ),
+            Depth(recurse=True),
+            FixedPoint("depth"),
+            Size(recurse=True),
+            FixedPoint("size"),
+            MinimumPoint(["depth", "size"], "optimization_loop"),
+        ],
+        do_while=lambda property_set: not property_set["optimization_loop_minimum_point"],
+    )
+)
+
+register_action(
+    DeviceDependentAction(
+        "BQSKitO2",
+        CompilationOrigin.BQSKIT,
+        PassType.OPT,
+        transpile_pass=lambda circuit: bqskit_compile(
+            circuit,
+            optimization_level=1 if os.getenv("GITHUB_ACTIONS") == "true" else 2,
+            synthesis_epsilon=1e-1 if os.getenv("GITHUB_ACTIONS") == "true" else 1e-8,
+            max_synthesis_size=2 if os.getenv("GITHUB_ACTIONS") == "true" else 3,
+            seed=10,
+            num_workers=1 if os.getenv("GITHUB_ACTIONS") == "true" else -1,
+        ),
+    )
+)
+
+register_action(
+    DeviceDependentAction(
+        "VF2PostLayout",
+        CompilationOrigin.QISKIT,
+        PassType.FINAL_OPT,
+        transpile_pass=lambda device: VF2PostLayout(target=device),
+    )
+)
+
+register_action(
+    DeviceDependentAction(
+        "TrivialLayout",
+        CompilationOrigin.QISKIT,
+        PassType.LAYOUT,
+        transpile_pass=lambda device: [
+            TrivialLayout(coupling_map=CouplingMap(device.build_coupling_map())),
+            FullAncillaAllocation(coupling_map=CouplingMap(device.build_coupling_map())),
+            EnlargeWithAncilla(),
+            ApplyLayout(),
+        ],
+    )
+)
```

Comment:
Might as well remove this to reduce the action space. In practically no circumstances will this be the solution to choose.